lib/queues.js

```diff
@@ -611,5 +610,22 @@ module.exports = class Queues {
     // Lazy load configuration.
     return this._ironium.configPromise;
   }
+
+  // Returns the maximum amount of jobs a queue should try to reserve.
+  // Every queue gets a share of reservation capacity to better deal
+  // with load spikes across queues.
+  getReserveCapacity() {
+    return this.configPromise
+      .then(config => {
+        const totalCapacity       = config.concurrency - this._activeJobs;
+        const maxCapacityPerQueue = Math.floor(totalCapacity / this._queues.size);
+        const capacity            = Math.min(totalCapacity, maxCapacityPerQueue);
```

Comment:
Reserving many jobs is cheap, but releasing them is expensive (one can be done in bulk, the other can't). So to minimize this, how about:

- Reserve up to 10% capacity on each call.
- Make the timeout in the reserve operation shorter, so the reserve capacity is evaluated more often (change from 30 seconds to say 5-10 seconds).